core/binding/src/tests/request_context.rs

```diff
@@ -0,0 +1,29 @@
+use protocol::traits::RequestContext;
+
+use crate::request_context::DefaultRequestContext;
+use crate::tests::sdk::mock_address;
+
+#[test]
+fn test_request_context() {
+    let mut ctx = DefaultRequestContext::new(
+        100,
+        8,
+        10,
+        mock_address(),
+        1,
+        "service_name".to_owned(),
+        "service_method".to_owned(),
+        "service_playload".to_owned(),
+    );
+
+    ctx.sub_cycles(8).unwrap();
+    assert_eq!(ctx.get_cycles_used(), 18);
+
+    assert_eq!(ctx.get_cycles_limit(), 100);
+    assert_eq!(ctx.get_cycles_price(), 8);
+    assert_eq!(ctx.get_caller(), mock_address());
+    assert_eq!(ctx.get_current_epoch_id(), 1);
+    assert_eq!(ctx.get_service_name(), "service_name".to_owned());
```

Comment:
I don't think can pass the test case. 

`ctx.get_service_name()` returns `&str`
`service_name".to_owned()` returns `String`

Did you run the `cargo test`?

Reply:
fixed.

I did run `cargo test`, but it did not complain